client/pseudonymsys.go

```diff
@@ -32,11 +33,13 @@ func (c *PseudonymsysClient) GenerateNym(userSecret *big.Int,
 	caCertificate *pseudonymsys.CACertificate) (
 	*pseudonymsys.Pseudonym, error) {
 	// new client needs to be created in each method to implicitly call server Run method:
-	genericClient, err := newGenericClient(c.endpoint)
+	genericClient, err := newGenericClient(c.conn)
```

Comment:
Please move `genericClient` struct initialization from
```
GenerateNym
ObtainCredential
TransferCredential
```
functions into the `NewPseudonymsysClient` constructor . You only need to open stream in each function.